internal/protocol/connection_id.go

```diff
@@ -27,58 +29,76 @@ func (c ArbitraryLenConnectionID) String() string {
 	return fmt.Sprintf("%x", c.Bytes())
 }
 
-// A ConnectionID in QUIC
-type ConnectionID []byte
-
 const maxConnectionIDLen = 20
 
+// A ConnectionID in QUIC
+type ConnectionID struct {
+	l uint8
```

Comment:
I think Go will pad this up to a word, so you could save a few bytes by having the array first.

Reply:
`unsafe.Sizeof(ConnectionID{})` is 21 in both cases, so it doesn't seem to make a difference.

This is the same change that @Joroppo suggested in https://github.com/lucas-clemente/quic-go/issues/3527#issuecomment-1229536697.